File: projects/order-creation/core/src/main/java/com/robinfood/core/mappers/GroupMappers.java
package com.robinfood.core.mappers;

import com.robinfood.core.dtos.transactionrequestdto.GroupDTO;
import com.robinfood.core.entities.GroupEntity;

import static kotlin.collections.CollectionsKt.map;

public final class GroupMappers {

    private GroupMappers() {
        throw new IllegalStateException("Utility class");
    }

    public static GroupEntity toGroupEntity(GroupDTO groupDTO) {
        if (groupDTO.getPortions().isEmpty()) {
            return null;
        }
        return new GroupEntity(
                groupDTO.getId(),
                map(groupDTO.getPortions(), PortionMappers::toPortionEntity),
                groupDTO.getSku()
        );
    }
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/mappers/GroupMappers.kt
package com.robinfood.core.mappers

import com.robinfood.core.dtos.transactionrequest.GroupDTO
import com.robinfood.core.entities.transactionrequest.GroupEntity

fun GroupDTO.toGroupEntity(): GroupEntity {
    return GroupEntity(
        id = id,
        name = name,
        portions = portions.map { portionDTO -> portionDTO.toPortionEntity() },
        sku = sku
    )
}